backend/src/routes/api/mlmd/index.ts

```diff
@@ -0,0 +1,78 @@
+import { FastifyRequest } from 'fastify';
+import httpProxy from '@fastify/http-proxy';
+import { V1Service } from '@kubernetes/client-node';
+import { KubeFastifyInstance } from '../../../types';
+import { isK8sStatus, passThroughResource } from '../k8s/pass-through';
+import { DEV_MODE } from '../../../utils/constants';
+import { createCustomError } from '../../../utils/requestUtils';
+
+const getParam = <F extends FastifyRequest<any, any>>(req: F, name: string) =>
+  (req.params as { [key: string]: string })[name];
+
+const setParam = (req: FastifyRequest, name: string, value: string) =>
+  ((req.params as { [key: string]: string })[name] = value);
+
+const serviceNotFound = (name: string, message = '') =>
+  createCustomError(
+    'Not Found',
+    `Service ds-pipeline-metadata-envoy-${name} not found.${message ? ` ${message}` : ''}`,
+    404,
+  );
+
+export default async (fastify: KubeFastifyInstance): Promise<void> => {
+  fastify.register(httpProxy, {
+    upstream: '',
+    prefix: '/:namespace/:name',
+    rewritePrefix: '',
+    replyOptions: {
+      // preHandler must set the `upstream` param
+      getUpstream: (request) => getParam(request, 'upstream'),
+    },
+    preHandler: (request, _, done) => {
+      const kc = fastify.kube.config;
+      const cluster = kc.getCurrentCluster();
+
+      // see `prefix` for named params
+      const namespace = getParam(request, 'namespace');
+      const name = getParam(request, 'name');
+
+      // retreive the metadata envoy service
+      passThroughResource<V1Service>(fastify, request, {
+        url: `${cluster.server}/api/v1/namespaces/${namespace}/services/ds-pipeline-metadata-envoy-${name}`,
```

Comment:
You're potentially introducing an SSRF vulnerability here as you're directly building the URL from data provided by the client, without first validating it.

Reply:
This call goes to the k8s API server only based on `cluster.server` which is not user input. The API server the handles whether the user has access to the resource.

The `params` are parsed by `http-proxy` matching a specific segment of the incoming URL. I am assuming that the `http-proxy` parser here does not allow inclusion of `/` within the param value but I didn't explicitly test that. It shouldn't though, otherwise that would defeat the purpose of the prefix path parser.

@alexcreasy Is there something else specific that you're referring to?